quinn/src/work_limiter.rs

```diff
@@ -0,0 +1,235 @@
+use std::time::{Duration, Instant};
+
+/// Limits the amount of time spent on a certain type of work in a cycle
+///
+/// The limiter works dynamically: For a sampled subset of cycles it measures the
+/// the time that is approximately required for fulfilling 1 work item, and
+/// calculates the amount of allowed work items per cycle.
+/// The estimates are smoothed over all cycles where the exact duration is measured.
+///
+/// In cycles where no measurement is performed the previously determined work limit
+/// is used.
+///
+/// For the limiter the exact definition of a work item does not matter.
+/// It could for example track the amount of transmitted bytes per cycle,
+/// or the amount of transmitted datagrams per cycle.
+/// It will however work best if the required time to complete a work item is
+/// constant.
+#[derive(Debug)]
+pub struct WorkLimiter {
+    /// Whether to measure the required work time, or to use the previous estimates
+    mode: LimiterMode,
```

Comment:
Let's make these field names a bit more concise, similar to most of the code. `LimiterMode` can just be called `Mode` (it's private anyway) and we can cut `work_item` from all field names. I think we can also do without the `_nanos` suffix, keeping as documentation only.

Reply:
I made most of them shorter. I kept `_nanos` since this is really something that I find immensly useful when looking at foreign code where the unit isn't clear, and have to point out in reviews constantly.

Also please keep in mind that there is a low downside of having longer names as long as they don't span the full line. Code readability wins in most cases by having them - and code is more often read than written..